sylt/src/test.rs

```diff
@@ -0,0 +1,258 @@
+        std::fs::read_dir(directory).expect(&format!("There is no such directory: {:?}", directory))
+    {
+        let path = entry.unwrap().path();
+        let file_name = path.file_name().unwrap().to_str().unwrap();
+
+        if file_name.starts_with("_") {
+            continue;
+        }
+
+        if path.is_dir() {
+            tests.extend(find_and_parse_tests(&path));
+        } else {
+            if !file_name.ends_with(".sy") {
+                continue;
+            }
+            let settings = parse_test_settings(std::fs::read_to_string(&path).unwrap());
+            tests.push(TestSettings { path: path, ..settings });
+        }
+    }
+    tests
+}
+
+fn run_test<R>(reader: R, settings: &TestSettings) -> bool
+where
+    R: Fn(&Path) -> Result<String, Error>,
+{
+    use std::process::{Command, Stdio};
+
+    let mut args = crate::Args::default();
+    args.args = vec![settings.path.to_str().unwrap().to_string()];
+    args.verbosity = if settings.print { 1 } else { 0 };
+
+    let mut child = Command::new("lua")
+        .stdin(Stdio::piped())
+        .stderr(Stdio::piped())
+        .stdout(Stdio::piped())
+        .spawn()
+        .expect(&format!("Failed to start: {:?}", settings.path));
+
+    let mut stdin = child.stdin.take().unwrap();
+    let res = crate::compile_with_reader_to_writer(&args, reader, &mut stdin);
+
+    drop(stdin); // Close stdin so the child can do its thing.
+
+    if let Err(errs) = res.clone() {
+        if compare_errors(errs.as_slice(), settings.errors.as_slice()) {
+            return true;
+        }
+        eprintln!("\n=== {:?} - failed in compiler", settings.path);
+        write_errors(&errs, &settings.errors);
+        return false;
+    }
+
+    let output = child.wait_with_output().unwrap();
+    // HACK(ed): Status is always 0 when piping to STDIN, atleast on my version of lua,
+    // so we check stderr - which is a bad idea.
+    let stderr = String::from_utf8_lossy(&output.stderr);
+    let stdout = String::from_utf8_lossy(&output.stdout);
+    let errs = if !(output.status.success() && stderr.is_empty()) {
+        vec![Error::RuntimeError]
+    } else {
+        Vec::new()
+    };
+
+    if !compare_errors(errs.as_slice(), settings.errors.as_slice()) {
+        eprintln!("\n=== {:?} - failed in runtime", settings.path);
+        if !stdout.is_empty() {
+            eprintln!("= STDOUT =\n{}", stdout);
+        }
+        if !stderr.is_empty() {
+            eprintln!("= STDERR =\n{}", stderr);
+        }
+        write_errors(&errs, &settings.errors);
+        return false;
+    }
+
+    true
+}
+
+#[test]
+fn program_tests() {
+    use crate::{formatter::format, read_file};
+    use std::env::set_current_dir;
+    use std::io::Write;
+    set_current_dir("../").unwrap();
+    let tests = find_and_parse_tests(Path::new("tests/"));
+    let mut failed = Vec::new();
+
+    writeln!(
+        std::io::stdout().lock(),
+        "= RUNNING {} TESTS =",
+        tests.len()
+    )
```

Comment:
```suggestion
    writeln!(
        std::io::stdout().lock(),
        "= RUNNING {} PROGRAM TESTS =",
        tests.len()
    )
```
or something along those lines. we have other tests in addition to these